### ISSUE TITLE
Bug in function download_label

### ISSUE THREAD

**Issue by manigalati** [Project-MONAI/MONAILabel#1014]
Hello! :)
I ran into the following bug while working with the plugin of monailabel in slicer.

**Describe the bug**
The application always fails at loading original labels, with the following message appearing in the log:
"Original label not found ... "

**To Reproduce**
After running batch inference of any model, original labels are created inside the server. When trying to fetch the next sample from the server in slicer, original labels never appear.

**Solution**
I debugged the code myself. The code always raises an exception that enters inside the catch block at line 1245 of file 
https://github.com/Project-MONAI/MONAILabel/blob/main/plugins/slicer/MONAILabel/MONAILabel.py 

The reason is to be found at line 173 of https://github.com/Project-MONAI/MONAILabel/blob/main/plugins/slicer/MONAILabel/MONAILabelLib/client.py

Here, the method getheader is invoked on the response object only after being read at line 357 of the same file. But response.read() turns the response from being an http response object to be a simple bytes object encapturing the content of the response. On a bytes object, getheader cannot be called, and this is what triggers an exception.

I hope I have been clear enough, so that my message can be useful to solve the issue quickly.

Best regards,
Francesco

**Comment by diazandr3s** [comment]
Hi @manigalati 

Thanks for raising this issue.

> The application always fails at loading original labels, with the following message appearing in the log:
> "Original label not found ... "

By default, MONAI Label searches in the labels/original folder for the predicted label for the current volume (https://github.com/Project-MONAI/MONAILabel/blob/main/plugins/slicer/MONAILabel/MONAILabel.py#L1234-L1245). If it doesn't exist, it will show that warning.

See the following image the flag `Original Labels` is activated. 

![Screenshot from 2022-09-27 00-14-15](https://user-images.githubusercontent.com/11991079/192397379-b02ac88a-48e5-47e3-a468-c1ed505dd30d.png)

You can get to this in Slicer on Edit -> Application Settings

> After running batch inference of any model, original labels are created inside the server. When trying to fetch the next sample from the server in slicer, original labels never appear.

How did you perform batch inference? Did you use the REST API on the web browser? Have you checked that the predictions are being generated on labels/original folder?

Hope that helps

**Comment by SachidanandAlle** [comment]
@tangy5 can you please check.. may be it needs something to be fixed on 3D slicer side..

**Comment by tangy5** [comment]
> @tangy5 can you please check.. may be it needs something to be fixed on 3D slicer side..

Good, I will look into this.

**Comment by diazandr3s** [comment]
I'm sorry, I've completely missed part of the text here:

> Here, the method getheader is invoked on the response object only after being read at line 357 of the same file. But response.read() turns the response from being an http response object to be a simple bytes object encapturing the content of the response. On a bytes object, getheader cannot be called, and this is what triggers an exception.

This means this we shouldn't use the .read method [here](https://github.com/Project-MONAI/MONAILabel/blob/main/plugins/slicer/MONAILabel/MONAILabelLib/client.py#L357) or [here](https://github.com/Project-MONAI/MONAILabel/blob/main/plugins/slicer/MONAILabel/MONAILabelLib/client.py#L354)?

**Comment by tangy5** [comment]
Hi @manigalati @diazandr3s , investigation from my side, the issue do exists. And @manigalati 's point is correct, the response.getheader() is not working after response.read().

I've created a solution here: https://github.com/Project-MONAI/MONAILabel/pull/1031
To return and pass response.getheader(content_disposition) in the send_response. 

I tested with a spleen segmentation task, it works normal. 

Let's see if we like this solution.
